src/filters.py

```diff
@@ -7,37 +7,32 @@
 def charswap_filter(word: str) -> list:
             
     #STEP 1 - initial replacements with reference to Leet system. Capitalized letters do not matter in this filter
-    lowercase = word.lower()
-    
+
     for combination, letter in SourceDictionaries.POSSIBLE_SWAPS.items():
-        if combination in lowercase:
-            lowercase = lowercase.replace(combination, letter)
+        if combination in word:
+            word = word.replace(combination, letter)
     
     possible_profanities = list()
-    possible_profanities.append(lowercase)
-    #STEP 2 - ambiguity check. f.e '4' can mean both 'a' or 'h' according to Leet so 2 variations are created instead of 1.
+    possible_profanities.append(word)
+    
+    # STEP 2 - ambiguity check. f.e '4' can mean both 'a' or 'h' according to Leet so 2 variations are created instead of 1.
     for ambiguity, letters in SourceDictionaries.AMBIGUITIES.items():
-        if ambiguity in lowercase:
+        if ambiguity in word:
             for letter in letters:
                 temp = list()
                 for profanity in possible_profanities:
                     if ambiguity in profanity:
```

Comment:
count function returns `-1` if element does not exist inside the list. Therefore, this if statement is redundant because of line underneath it: `if profanity.count(ambiguity) > 1: `

Reply:
No, it returns 0. Deletion of `if ambiguity in profanity: ` changes the way of that combinations are made. Idk if it's for good or worse but certainly its not redundant.

Without the above line: 
['kurv4', 'kurva', 'kurvh', 'kurva', 'kuru4', 'kurua', 'kuruh', 'kurua', 'kurw4', 'kurwa', 'kurwh', 'kurwa', 'kuru4', 'kurua', 'kuruh', 'kurua']
With:
['kurv4', 'kurva', 'kurvh', 'kuru4', 'kurua', 'kuruh', 'kurw4', 'kurwa', 'kurwh']

Reply:
But if you want a visual example, just take a look at the first list with variations of possible "kurwa" (Without the above line). You can clearly see 'kurua' occuring thrice, both 'kuruh' and 'kuru4' occuring twice etc...